app/controllers/roles_controller.rb

```diff
@@ -34,6 +33,7 @@ def create
     @role = role_from_form
 
     if @role.save
+      @role.set_taxonomies(taxonomy_params!) if cloning?
```

Comment:
Business logic, the model should take care of that (feel free to move taxonomy stuff to a concern if that's cleaner)